### PR TITLE
UPI Table converter to google.protobuf.Struct

### PR DESCRIPTION
**Pull request by pradithya**
`RouterLog` and `PredictionLog` use `google.protobuf.Struct` for publishing the UPI Table. This PR introduces the converter for converting UPI Table into the corresponding `google.protobuf.Struct`.  Note that `TableToStruct` accept `schemaVersion` which is determined by [table_schema_version](https://github.com/caraml-dev/universal-prediction-interface/blob/ce0f30be060f2dfeeba03574d53bbe92a514bc20/proto/caraml/upi/v1/prediction_log.proto#L33) and this PR add implementation for `TableSchemaV1` . 

Example usage:
```go
package pkg

import (
	"fmt"

	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
	"github.com/caraml-dev/universal-prediction-interface/pkg/converter"
)

func main() {
	var upiTable upiv1.Table
	st, err := converter.TableToStruct(&upiTable, converter.TableSchemaV1)
	if err != nil {
		// conversion failed
	}

	fmt.Printf("%v", st)
}
```

### CODE REVIEW

File: pkg/converter/struct_v1.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	"fmt"
+
+	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	IntType     = "INT64"
+	NumberType  = "FLOAT64"
+	StringType  = "STRING"
```
Comment (leonlnj) [inline pkg/converter/struct_v1.go:13]: for my understanding where does `INT64 FLOAT64 STRING` come from? 

Also was confused abit when I saw `NumberType` but guess its not outputted anywhere

Comment (pradithya) [reply pkg/converter/struct_v1.go:13]: > for my understanding where does INT64 FLOAT64 STRING come from?

This is coming from the BQ column type. It'll be used later on during query time from the log table. I think it's better to rename the const to reflect its value better.

**Comment by leonlnj** [comment]
Would there be a use case for struct to table? Left a small comment, the rest LGTM

Seeing the slight variation of types confused me for abit when review 🤣 
UPI Type
image

ProtoBq
image

Google structpb
image

**Review by leonlnj — APPROVED** [review]

File: pkg/converter/struct_v1.go (hunk 1 above)
Comment (leonlnj) [reply pkg/converter/struct_v1.go:13]: ah, I thought the infer schema will be FLOAT (happen to have a ss below), number is only used in structpb I think.

Was just a curious question if there was something that I missed, thanks!

Comment (pradithya) [reply pkg/converter/struct_v1.go]: Oh you are right. Let me update it accordingly .

**Pull request merged**